wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java

```diff
@@ -170,8 +171,13 @@ public Connection connect(
       final boolean isInitialConnection,
       final JdbcCallable<Connection, SQLException> connectFunc)
       throws SQLException {
-
-    ConnectionProvider connProvider = this.connProviderManager.getConnectionProvider(driverProtocol, hostSpec, props);
+    ConnectionProvider connProvider;
+    if (PropertyDefinition.IS_AUXILIARY_CONNECTION.getBoolean(props))  {
```

Comment:
We already have forceConnect pipeline for that.

Reply:
This is my understanding from our discussions about ConnectionService: the problem with forceConnect is that it results in shared plugin services between connections, when it would be cleaner if each connection were separate and had their own plugin services. The main difference between forceConnect and createAuxiliaryConnection is that connections created with createAuxiliaryConnection do not share plugin services. Let me know if I am missing something. Also, see [this todo](https://github.com/aws/aws-advanced-jdbc-wrapper/blob/storage-service/wrapper/src/main/java/software/amazon/jdbc/util/connection/ConnectionServiceImpl.java#L65), I'm not sure how to make sure the new connection does not share a connection provider or target driver dialect.